pages/reset-password/[token].js

```diff
@@ -0,0 +1,148 @@
+import React, { useRef, useState } from "react";
+import propsWithContainer from "../../src/middleware/propsWithContainer";
+import Button from "../../src/components/Button";
+import ErrorSummary from "../../src/components/ErrorSummary";
+import FormGroup from "../../src/components/FormGroup";
+import { GridRow, GridColumn } from "../../src/components/Grid";
+import Heading from "../../src/components/Heading";
+import Input from "../../src/components/Input";
+import Label from "../../src/components/Label";
+import Layout from "../../src/components/Layout";
+import Form from "../../src/components/Form";
+import Router from "next/router";
+
+const ResetPassword = ({ email, tokenError }) => {
+  const [errors, setErrors] = useState([]);
+  if (tokenError) {
+    errors.push({
+      id: "token-error",
+      message: tokenError,
+    });
+  }
+
+  const passwordRef = useRef();
+  const confirmPasswordRef = useRef();
+
+  const hasError = (field) =>
+    errors.find((error) => error.id === `${field}-error`);
+
+  const errorMessage = (field) => {
+    const error = errors.filter((err) => err.id === `${field}-error`);
+    return error.length === 1 ? error[0].message : "";
+  };
```

Comment:
See similar comment here: https://github.com/madetech/nhs-virtual-visit/pull/743#discussion_r557470422

Reply:
Refactored to use helper methods